pkg/cmd/extension/manager.go

```diff
@@ -249,6 +250,64 @@ func (m *Manager) installDir() string {
 	return filepath.Join(m.dataDir(), "extensions")
 }
 
+func (m *Manager) Create(name string) error {
+	exe, err := m.lookPath("git")
+	if err != nil {
+		return err
+	}
+
+	err = os.Mkdir(name, 0755)
+	if err != nil {
+		return err
+	}
+
+	initCmd := m.newCommand(exe, "init", "-b", "main", "--quiet", name)
+	err = initCmd.Run()
+	if err != nil {
+		return err
+	}
+
+	fileTmpl := heredoc.Docf(`
+		#!/bin/bash
+		set -e
+		echo "Hello %[1]s"
+
+		# Examples to help get started:
+
+		# Determine if an executable is in the PATH
+		# if [ "$(which go)" = "" ]; then
+		#   echo "go not found in PATH"
+		#   exit 1
+		# fi
+
+		# Pass arguments through to a new process
+		# exec gh issue list "$@" -R cli/cli
+
+		# Using the gh api command to retrieve and format information
+		# QUERY='
+		#   query($endCursor: String) {
+		#     viewer {
+		#       repositories(first: 100, after: $endCursor) {
+		#         nodes {
+		#           nameWithOwner
+		#           stargazerCount
+		#         }
+		#       }
+		#     }
+		#   }
+		# '
+		# TEMPLATE='
+		#   {{- range $repo := .data.viewer.repositories.nodes -}}
+		#     {{- printf "name: %[2]s - stargazers: %[3]s\n" $repo.nameWithOwner $repo.stargazerCount -}}
+		#   {{- end -}}
+		# '
+		# exec gh api graphql --template="${TEMPLATE}" --paginate -f query="${QUERY}"
+	`, name, "%s", "%v")
+	filePath := filepath.Join(name, name)
+	err = ioutil.WriteFile(filePath, []byte(fileTmpl), 0755)
```

Comment:
Just had a thought: Windows doesn't have a concept of executable files. When this file gets added to git, will it retain its executable bit? If not, this could be a workaround on Windows:
```
git add file_name
git update-index --chmod=+x file_name
```

Reply:
I asked @vilmibm to help me check on this and the file created was executable on Windows. I don't know if @vilmibm confirmed that when it was added to git it maintained the executable property though.

Reply:
This is unfortunately the trickiest part of publishing an extension on Windows, so if it turns out that it's needed, I would vote to add the `git add && update-index` calls there (or on either platform—it would be a no-op on other platforms anyway!)

Reply:
i just created the extension and ran the `gh-test` file; i didn't try adding to git

Reply:
I went ahead and added `git add ... --chmod=+x` command to the flow.